### ISSUE TITLE
Expose `AnimationTrackEditor` and `AnimationTrack*` related classes

### ISSUE THREAD

**Issue by AnidemDex** [godotengine/godot-proposals#3468]
### Describe the project you are working on

Textalog - An event manager with an integrated dialogue system plugin for Godot

### Describe the problem or limitation you are having in your project

My project started as a fork of Dialogic, but suddenly I noticed that I didn't want to replicate its timeline behaviour (wich is only one timeline per dialogue and consecutive events in only one track).

I decided to move the timeline system to a _real timeline_, using the integrated Animation Editor in Godot (maybe is named Timeline Editor?) to integrate consecutive events in time, with multiple tracks with multiple events in parallel, only to discover that all related nodes, methods and resurces are not exposed, making it difficulty to replicate and/or modify the existing one.

Animation Editor node is `AnimationTrackEditor` class. You can't instance it directly and can't instance it through `ClassDB`, neither any [animation editor plugin](https://github.com/godotengine/godot/blob/3.3/editor/animation_track_editor_plugins.h)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Exposing animation editor related classes will help those developers that want to modify or improve the current animation editor. For example, adding offsets to tracks or creating clips making groups of tracks and reusing those clips later (similar to Unity's timeline track clips). #3212 and https://github.com/godotengine/godot-proposals/issues/357 are other examples that may be related. (I think, exposing those classes give the ability to plugin makers to modify the timeline according to what they want) 

This also can help in those games that uses a kind of track editor (like music games custom levels)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I have no idea how to implement a solution in the source code. I only know that related files are probably

https://github.com/godotengine/godot/blob/3.3/scene/resources/animation.h#L66
and
https://github.com/godotengine/godot/blob/ce11f8f36b41e1d6c881d71bdacb82250bbab0e3/editor/plugins/animation_player_editor_plugin.h

AnimationPlayer has already some exposed methods but the class itself it's hard to get in the editor (navigating throguh the editor tree using magic tricks with `get_node`) and doesn't let you modify it without giving you a headache. And you can't use it in any game, just in the editor.

_(Tried to make an embedded code snippet but I failed)_

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, it there's no simple workaround. Currently I've to reimplement the whole animation editor manually, modifying the Animation resource through code per track and frames.

### Is there a reason why this should be core and not an add-on in the asset library?

Non exposed classes can't be accesed in code unless we recompile the engine and register and expose those classes

**Comment by Creta5164** [comment]
Here, I came back to here when I was looking for work around about manipulating `AnimationPlayerEditor`, I want to everyone consider take this slightly more prioritize this then before.

To put it in terms of my actual situation, I currently wish I could use the transform recording feature that appears in the toolbar built into Godot for the nodes and properties I want. However, since that's not currently possible, I ended up writing code with very long boilerplate, as shown below.

```gdscript
static func insert_animation_key(target: Object, property_name: String, value: Variant, advance: bool = false) -> void:
    
    if !is_instance_valid(instance):
        return
    
    #if instance._is_inserting_animation_key:
    #    await instance._animation_key_inserted
    
    instance._is_inserting_animation_key   = true
    instance._halt_selection_change_action = true
    
    var editor_undo_redo := EditorInterface.get_editor_undo_redo()
    var editor_inspector := EditorInterface.get_inspector()
    var editor_selection := EditorInterface.get_selection()
    
    #if editor_undo_redo.is_committing_action():
    #    await editor_undo_redo.history_changed
    
    var previously_selected := editor_selection.get_selected_nodes()
    var previously_edited   := editor_inspector.get_edited_object()
    
    #Editor selection update has deferred timing.
    if !previously_selected.has(target):
        
        editor_selection.clear()
        editor_selection.add_node(target)
        
        await editor_selection.selection_changed
        
        pass
    
    if previously_edited != target:
        editor_inspector.edit(target)
    
    _perform_insert_animation_key.call_deferred(property_name, value, advance)
    
    #await editor_inspector.property_keyed
    
    instance._halt_selection_change_action = false
    instance._is_inserting_animation_key   = false
    
    #instance._animation_key_inserted.emit.call_deferred()
    
    pass

static func _perform_insert_animation_key(property_name: String, value: Variant, advance: bool) -> void:
    
    const SETTINGS_CONFIRM_INSERT_TRACK := "editors/animation/confirm_insert_track"
    
    var editor_inspector := EditorInterface.get_inspector()
    
    var settings := EditorInterface.get_editor_settings()
    
    var previous_settings_value := settings.get(SETTINGS_CONFIRM_INSERT_TRACK)
    
    settings.set(SETTINGS_CONFIRM_INSERT_TRACK, false)
    
    editor_inspector.property_keyed.emit(property_name, value, advance)
    
    settings.set(SETTINGS_CONFIRM_INSERT_TRACK, previous_settings_value)
    
    pass
```
\*Note that some comments are went to try some experiment for inserting keys by depending on how `Animation*Editor` classes implements how are they manage them in Godot editor.

It works, but I discovered it very easily abuses the UndoRedo history, so I stopped working on this feature. I felt stressed for a week, wondering if it was really necessary to go to such lengths.

<img width="424" height="934" alt="Image" src="https://github.com/user-attachments/assets/8dcc176c-ad03-4395-bfd8-e0daac280777" />

What I wanted, and my thoughts on this, are as follows:
- I wanted to build an integrated workflow using the infrastructure provided by Godot.
- As discussed in the issue body, the animation side is undergoing significant changes recently, as seen in the development blog (I'll find and add the link later), but the scale seems to indicate it will take a long time to complete.
- However, building my own infrastructure or features would likely compromise Godot's standards significantly, making me hesitant.

Alongside #13097, if contributors or developers who prefer editor tooling agree with these issues, I'd be grateful if you could give both issues a 👍...

---

Edit: This problem can be workaround more simpler way if you can grab `NodePath` of these editor interface nodes, [watch this video](https://youtu.be/ZGIZavfnbX4), but please expect this workaround will depends Godot version, I suppose.
